projects/transformers/run_utils.py

```diff
@@ -459,6 +484,286 @@ def preprocess_function(examples):
     return tokenized_datasets
 
 
+def squad_prepare_features_factory(  # noqa: C901
+                                   split,
+                                   data_args,
+                                   tokenizer,
+                                   tokenizer_kwargs,
+                                   question_column_name=None,
+                                   context_column_name=None,
+                                   answer_column_name=None,
+                                   pad_on_right=None,
+                                   ):
+    """
+    splits can be train / val / and beam search ca be yes or no
+    """
+
+    key1 = question_column_name if pad_on_right else context_column_name
+    key2 = context_column_name if pad_on_right else question_column_name
+
+    def pre_pre_process(examples):
+
+        tokenized_examples = tokenizer(
+            examples[key1], examples[key2], **tokenizer_kwargs)
+
+        return tokenized_examples
+
+    if split == "train":
+
+        def preprocess_function(examples):
+
+            examples[question_column_name] = [q.lstrip() for q in examples[question_column_name]]  # noqa: E501
```

Comment:
Although I strongly disagree with the 80 chars per line requirement as I believe it incentivizes poor variable naming (a heated discussion in our group), we should avoid leaving tons of flake 8 exceptions in the code. For example an easy fix for this would be (if it fits the line char requirements)

```
examples[question_column_name] = [
    q.lstrip() for q in examples[question_column_name]
]
```